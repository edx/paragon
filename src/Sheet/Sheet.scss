.pgn__sheet-skrim {
  width: 100%;
  height: 100%;
  background-color: rgba(196,196,196,0.5);
  position: fixed;
  top: 0px;
  left: 0px;
  font-size: 18px;
  &.hidden {
    display: none;
  }
}
.pgn__sheet-component {
  position: fixed;
  padding: 20px;
  background-color: white;
  &.pgn__sheet__dark {
    background-color: rgb(0,38,43);
    color: rgb(242,240,239);
  }
  &.bottom {
    bottom: 0px;
    box-shadow: 0px -4px 10px 0px rgba(0,0,0,0.15);
    box-shadow: 0px -8px 16px 0px rgba(0,0,0,0.15);
  }
  &.top {
    top: 0px;
    box-shadow: 0px 4px 10px 0px rgba(0,0,0,0.15);
    box-shadow: 0px 8px 16px 0px rgba(0,0,0,0.15);
  }
  &.left {
    left: 0px;
    box-shadow: 4px 0px 10px 0px rgba(0,0,0,0.15);
    box-shadow: 8px 0px 16px 0px rgba(0,0,0,0.15);
  }
  &.right {
    right: 0px;
    box-shadow: -4px 0px 10px 0px rgba(0,0,0,0.15);
    box-shadow: -8px 0px 16px 0px rgba(0,0,0,0.15);
  }

  &.bottom, &.top {
    width: 100%;
    left: 0px;
  }
  &.left, &.right {
    height: 100%;
    top: 0px;
  }
}
