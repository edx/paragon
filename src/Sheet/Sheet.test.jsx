import React from 'react';
import renderer from 'react-test-renderer';
import Sheet, { POSITIONS, VARIANTS } from './index';

/* eslint-disable react/prop-types */
jest.mock('./SheetContainer', () => (props) => {
  const { children, ...otherProps } = props;
  return (
    <sheet-container {...otherProps}>
      {children}
    </sheet-container>
  );
});

const testContent = (<div className="sheet-content">Hi</div>);

const renderJSON = (jsxContent) => renderer.create(jsxContent).toJSON();

describe('<Sheet />', () => {
  describe('snapshots', () => {
    test('default args snapshot: bottom, show, light', () => {
      const el = renderJSON(<Sheet>{testContent}</Sheet>);
      expect(el).toMatchSnapshot();
    });

    test('blocking, left snapshot', () => {
      expect(
        renderJSON(<Sheet blocking position={POSITIONS.left} />),
      ).toMatchSnapshot();
    });

    test('dark, right snapshot', () => {
      expect(
        renderJSON(<Sheet position={POSITIONS.right} variant={VARIANTS.dark} />),
      ).toMatchSnapshot();
    });
  });

  it('returns empty render iff show is false', () => {
    expect(renderJSON(<Sheet show={false} />)).toEqual(null);
    expect(renderJSON(<Sheet />)).not.toEqual(null);
  });
});
