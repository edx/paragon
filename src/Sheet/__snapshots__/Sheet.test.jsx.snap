// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Sheet /> snapshots blocking, left snapshot 1`] = `
<sheet-container>
  <div
    className="pgn__sheet-skrim"
    onClick={[Function]}
    role="presentation"
  />
  <div
    aria-atomic="true"
    aria-live="polite"
    className="pgn__sheet-component pgn__sheet__light left"
    role="alert"
  >
    <div
      className="pgn__sheet-content"
    />
  </div>
</sheet-container>
`;

exports[`<Sheet /> snapshots dark, right snapshot 1`] = `
<sheet-container>
  <div
    className="pgn__sheet-skrim hidden"
    onClick={[Function]}
    role="presentation"
  />
  <div
    aria-atomic="true"
    aria-live="polite"
    className="pgn__sheet-component pgn__sheet__dark right"
    role="alert"
  >
    <div
      className="pgn__sheet-content"
    />
  </div>
</sheet-container>
`;

exports[`<Sheet /> snapshots default args snapshot: bottom, show, light 1`] = `
<sheet-container>
  <div
    className="pgn__sheet-skrim hidden"
    onClick={[Function]}
    role="presentation"
  />
  <div
    aria-atomic="true"
    aria-live="polite"
    className="pgn__sheet-component pgn__sheet__light bottom"
    role="alert"
  >
    <div
      className="pgn__sheet-content"
    >
      <div
        className="sheet-content"
      >
        Hi
      </div>
    </div>
  </div>
</sheet-container>
`;
